Check service existence before creation

qingc opened the pull request:
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
<!--- Please provide the related command with az {command} if you can, so that we can quickly route to the related person to review. --->


### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).


### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

yonzhan commented:
spring

yuwzho commented on src/spring/azext_spring/custom.py:
```diff
@@ -85,6 +85,13 @@ def _update_application_insights_asc_create(cmd,
                         monitoring_setting_resource=monitoring_setting_resource)
 
 
+def _check_spring_instance_existed(cmd, client, resource_group, name, location, **_):
```
NIT: This can be put into validator before running the command.

yuwzho commented on src/spring/azext_spring/custom.py:
```diff
@@ -85,6 +85,13 @@ def _update_application_insights_asc_create(cmd,
+    availability_parameters = models.NameAvailabilityParameters(type="Microsoft.AppPlatform/Spring", name=name)
+    name_availability = client.services.check_name_availability(location, availability_parameters)
+    if not name_availability.name_available and name_availability.reason == "AlreadyExists":
+        raise ClientRequestError("Service instance '{}' under resource group '{}' is already existed in region '{}', cannot create again.".format(name, resource_group, location))
```
```suggestion
        raise ClientRequestError("Service instance '{}' under resource group '{}' is already existed in region '{}', cannot be created again.".format(name, resource_group, location))
```

qingc commented:
@zhoxing-ms Could you help to review this PR? I have addressed comments from our team member and passed all the test. It's ready for review now. Many thanks!
